refactor(core): deprecate entryComponents (#33205)
angular/angular: aio/content/guide/deprecations.md

This includes both packages: `@angular/platform-webworker` and
`@angular/platform-webworker-dynamic`.

{@a entryComponents}
### `entryComponents` and `ANALYZE_FOR_ENTRY_COMPONENTS` no longer required

> Previously, the `entryComponents` array in the `NgModule` definition was used to tell the compiler, ahead of time, which components are going to be created and inserted dynamically. With Ivy this isn't a requirement anymore and the `entryComponents` array can be removed from existing module declarations. The same applies to the `ANALYZE_FOR_ENTRY_COMPONENTS` injection token.

Reviewer: ```suggestion
Previously, the `entryComponents` array in the `NgModule` definition was used to tell the compiler which components would be created and inserted dynamically. With Ivy, this isn't a requirement anymore and the `entryComponents` array can be removed from existing module declarations. The same applies to the `ANALYZE_FOR_ENTRY_COMPONENTS` injection token.
```
